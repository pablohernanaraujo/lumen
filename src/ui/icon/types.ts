import type { ReactNode } from 'react';

import type { Theme } from '../../theme';

export type IconSize = keyof Theme['typography']['size'] | number;

export type IconFamily =
  | 'AntDesign'
  | 'Entypo'
  | 'EvilIcons'
  | 'Feather'
  | 'FontAwesome'
  | 'FontAwesome5'
  | 'Foundation'
  | 'Ionicons'
  | 'MaterialIcons'
  | 'MaterialCommunityIcons'
  | 'Octicons'
  | 'SimpleLineIcons'
  | 'Zocial';

export interface IconProps {
  name: string;
  family?: IconFamily;
  size?: IconSize;
  color?: string;
  style?: object;
  testID?: string;
  onPress?: () => void;
  accessible?: boolean;
  accessibilityLabel?: string;
  children?: ReactNode;
}
