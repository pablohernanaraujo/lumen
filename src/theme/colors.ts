export const colorTokens = {
  light: {
    background: '#FFFFFF',
    surface: '#F8F9FA',
    surfaceSecondary: '#F1F3F4',
    text: {
      primary: '#1A1A1A',
      secondary: '#6B7280',
      tertiary: '#9CA3AF',
      inverse: '#FFFFFF',
    },
    primary: {
      50: '#EFF6FF',
      100: '#DBEAFE',
      200: '#BFDBFE',
      300: '#93C5FD',
      400: '#60A5FA',
      500: '#3B82F6',
      600: '#2563EB',
      700: '#1D4ED8',
      800: '#1E40AF',
      900: '#1E3A8A',
      main: '#3B82F6',
      dark: '#1D4ED8',
      light: '#93C5FD',
    },
    secondary: {
      50: '#FFF7ED',
      100: '#FFEDD5',
      200: '#FED7AA',
      300: '#FDBA74',
      400: '#FB923C',
      500: '#F97316',
      600: '#EA580C',
      700: '#C2410C',
      800: '#9A3412',
      900: '#7C2D12',
      main: '#F97316',
      dark: '#C2410C',
      light: '#FDBA74',
    },
    success: {
      main: '#10B981',
      dark: '#059669',
      light: '#6EE7B7',
    },
    warning: {
      main: '#F59E0B',
      dark: '#D97706',
      light: '#FCD34D',
    },
    error: {
      main: '#EF4444',
      dark: '#DC2626',
      light: '#FCA5A5',
    },
    info: {
      main: '#3B82F6',
      dark: '#1D4ED8',
      light: '#93C5FD',
    },
    border: '#E5E7EB',
    borderLight: '#F3F4F6',
    borderDark: '#D1D5DB',
    overlay: 'rgba(0, 0, 0, 0.5)',
  },
  dark: {
    background: '#000000',
    surface: '#111111',
    surfaceSecondary: '#1F1F1F',
    text: {
      primary: '#FFFFFF',
      secondary: '#A1A1AA',
      tertiary: '#71717A',
      inverse: '#000000',
    },
    primary: {
      50: '#1E3A8A',
      100: '#1E40AF',
      200: '#1D4ED8',
      300: '#2563EB',
      400: '#3B82F6',
      500: '#60A5FA',
      600: '#93C5FD',
      700: '#BFDBFE',
      800: '#DBEAFE',
      900: '#EFF6FF',
      main: '#60A5FA',
      dark: '#2563EB',
      light: '#BFDBFE',
    },
    secondary: {
      50: '#7C2D12',
      100: '#9A3412',
      200: '#C2410C',
      300: '#EA580C',
      400: '#F97316',
      500: '#FB923C',
      600: '#FDBA74',
      700: '#FED7AA',
      800: '#FFEDD5',
      900: '#FFF7ED',
      main: '#FB923C',
      dark: '#EA580C',
      light: '#FED7AA',
    },
    success: {
      main: '#34D399',
      dark: '#10B981',
      light: '#A7F3D0',
    },
    warning: {
      main: '#FBBF24',
      dark: '#F59E0B',
      light: '#FEF3C7',
    },
    error: {
      main: '#F87171',
      dark: '#EF4444',
      light: '#FECACA',
    },
    info: {
      main: '#60A5FA',
      dark: '#3B82F6',
      light: '#BFDBFE',
    },
    border: '#374151',
    borderLight: '#4B5563',
    borderDark: '#1F2937',
    overlay: 'rgba(0, 0, 0, 0.8)',
  },
} as const;
