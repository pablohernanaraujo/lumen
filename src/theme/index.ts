export { colorTokens } from './colors';
export { makeStyles } from './make-styles';
export { ThemeProvider, useTheme } from './theme-provider';
export { baseTokens } from './tokens';
export type {
  BaseTokens,
  ColorTokens,
  EmphasisLevel,
  Mode,
  NamedStyles,
  StyleFunction,
  StylesOrFunction,
  Theme,
  ThemeColors,
  ThemeContextValue,
  ThemeProviderProps,
} from './types';
export {
  createButtonStyle,
  createCardStyle,
  createInputStyle,
  createShadow,
  createTextStyle,
  getColorOpacity,
  getContrastColor,
  isLightColor,
} from './utils';
