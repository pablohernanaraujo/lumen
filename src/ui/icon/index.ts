export { Icon } from './icon';
export type { IconFamily, IconProps, IconSize } from './types';
