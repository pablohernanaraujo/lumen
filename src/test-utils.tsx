import React, { type FC, type ReactElement } from 'react';
import {
  render,
  type RenderOptions,
  type RenderResult,
} from '@testing-library/react-native';

import { ThemeProvider } from './theme';
import type { Mode } from './theme/types';

interface AllTheProvidersProps {
  children: React.ReactNode;
  mode?: Mode;
}

const AllTheProviders: FC<AllTheProvidersProps> = ({
  children,
  mode = 'light',
}) => <ThemeProvider preference={mode}>{children}</ThemeProvider>;

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  mode?: Mode;
  wrapper?: React.ComponentType<{ children: React.ReactNode }>;
}

export const customRender = (
  ui: ReactElement,
  { mode = 'light', wrapper, ...renderOptions }: CustomRenderOptions = {},
): RenderResult => {
  const Wrapper: FC<{ children: React.ReactNode }> = ({ children }) => {
    const content = <AllTheProviders mode={mode}>{children}</AllTheProviders>;

    return wrapper
      ? React.createElement(wrapper, { children: content })
      : content;
  };

  return render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
};

// Re-export everything from React Native Testing Library
export * from '@testing-library/react-native';

// Override the default render with our custom one
export { customRender as render };

// Create snapshot matcher helper
export const createSnapshot = (
  component: ReactElement,
  mode: Mode = 'light',
): void => {
  const { toJSON } = customRender(component, { mode });
  expect(toJSON()).toMatchSnapshot();
};

// Create both light and dark mode snapshots
export const createDualSnapshots = (component: ReactElement): void => {
  describe('snapshots', () => {
    it('should render correctly in light mode', () => {
      createSnapshot(component, 'light');
    });

    it('should render correctly in dark mode', () => {
      createSnapshot(component, 'dark');
    });
  });
};

// Export TestWrapper for use in tests
export const TestWrapper: FC<{ children: React.ReactNode; mode?: Mode }> = ({
  children,
  mode = 'light',
}) => <AllTheProviders mode={mode}>{children}</AllTheProviders>;
