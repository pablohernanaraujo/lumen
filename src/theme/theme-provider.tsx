import React, {
  createContext,
  type FC,
  useCallback,
  useContext,
  useMemo,
  useState,
} from 'react';
import { useColorScheme } from 'react-native';

import { colorTokens } from './colors';
import { baseTokens } from './tokens';
import type {
  Mode,
  Theme,
  ThemeContextValue,
  ThemeProviderProps,
} from './types';

const ThemeContext = createContext<ThemeContextValue | null>(null);

export const ThemeProvider: FC<ThemeProviderProps> = ({
  children,
  preference,
  persistenceKey = '@lumen-theme',
}) => {
  const systemMode = useColorScheme() as Mode | null;
  const [userMode, setUserMode] = useState<Mode | null>(preference || null);

  const currentMode: Mode = userMode ?? systemMode ?? 'light';

  const theme = useMemo<Theme>(
    () => ({
      ...baseTokens,
      colors: colorTokens[currentMode],
      mode: currentMode,
    }),
    [currentMode],
  );

  const setMode = useCallback((mode: Mode): void => {
    setUserMode(mode);
  }, []);

  const toggleMode = useCallback((): void => {
    setUserMode(currentMode === 'light' ? 'dark' : 'light');
  }, [currentMode]);

  const contextValue = useMemo<ThemeContextValue>(
    () => ({
      theme,
      mode: currentMode,
      setMode,
      toggleMode,
    }),
    [theme, currentMode, setMode, toggleMode],
  );

  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = (): ThemeContextValue => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};
