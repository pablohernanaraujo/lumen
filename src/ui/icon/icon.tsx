/* eslint-disable complexity */
import React, { type ComponentType, type FC } from 'react';
import { TouchableOpacity } from 'react-native';
import AntDesign from 'react-native-vector-icons/AntDesign';
import Entypo from 'react-native-vector-icons/Entypo';
import EvilIcons from 'react-native-vector-icons/EvilIcons';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import Foundation from 'react-native-vector-icons/Foundation';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Octicons from 'react-native-vector-icons/Octicons';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';
import Zocial from 'react-native-vector-icons/Zocial';

import { useTheme } from '../../theme';
import type { IconFamily, IconProps } from './types';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const getIconComponent = (family: IconFamily): ComponentType<any> => {
  switch (family) {
    case 'AntDesign':
      return AntDesign;
    case 'Entypo':
      return Entypo;
    case 'EvilIcons':
      return EvilIcons;
    case 'Feather':
      return Feather;
    case 'FontAwesome':
      return FontAwesome;
    case 'FontAwesome5':
      return FontAwesome5;
    case 'Foundation':
      return Foundation;
    case 'Ionicons':
      return Ionicons;
    case 'MaterialIcons':
      return MaterialIcons;
    case 'MaterialCommunityIcons':
      return MaterialCommunityIcons;
    case 'Octicons':
      return Octicons;
    case 'SimpleLineIcons':
      return SimpleLineIcons;
    case 'Zocial':
      return Zocial;
    default:
      return MaterialIcons;
  }
};

export const Icon: FC<IconProps> = ({
  name,
  family = 'MaterialIcons',
  size = 'md',
  color,
  style,
  testID,
  onPress,
  accessible = true,
  accessibilityLabel,
  children,
}) => {
  const { theme } = useTheme();

  const IconComponent = getIconComponent(family);

  const iconSize =
    typeof size === 'number' ? size : theme.typography.size[size];
  const iconColor = color || theme.colors.text.primary;

  const iconElement = (
    <IconComponent
      name={name}
      size={iconSize}
      color={iconColor}
      style={style}
      testID={testID}
      accessible={accessible}
      accessibilityLabel={accessibilityLabel || `${family} ${name} icon`}
    >
      {children}
    </IconComponent>
  );

  if (onPress) {
    return (
      <TouchableOpacity
        onPress={onPress}
        accessible={accessible}
        accessibilityLabel={accessibilityLabel || `${family} ${name} button`}
        accessibilityRole="button"
        testID={testID ? `${testID}-button` : undefined}
      >
        {iconElement}
      </TouchableOpacity>
    );
  }

  return iconElement;
};
