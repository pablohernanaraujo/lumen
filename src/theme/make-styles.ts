import { StyleSheet } from 'react-native';

import { useTheme } from './theme-provider';
import type { NamedStyles, StylesOrFunction, Theme } from './types';

export const makeStyles =
  <T extends NamedStyles<T>>(
    factory: StylesOrFunction<T>,
  ): (() => StyleSheet.NamedStyles<T>) =>
  (): StyleSheet.NamedStyles<T> => {
    const { theme } = useTheme();

    const styles =
      typeof factory === 'function'
        ? (factory as (theme: Theme) => T)(theme)
        : factory;

    return StyleSheet.create(styles);
  };
