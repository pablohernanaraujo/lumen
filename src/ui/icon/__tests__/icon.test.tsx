/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import { fireEvent } from '@testing-library/react-native';

import { createDualSnapshots, render } from '../../../test-utils';
import { makeStyles } from '../../../theme';
import { Icon } from '../icon';

const useTestStyles = makeStyles((theme) => ({
  marginStyle: {
    margin: theme.spacing.sm,
  },
}));

describe('Icon', () => {
  createDualSnapshots(<Icon name="home" testID="icon-1" />);

  describe('with different families', () => {
    const families = [
      'AntDesign',
      'Entypo',
      'EvilIcons',
      'Feather',
      'FontAwesome',
      'FontAwesome5',
      'Foundation',
      'Ionicons',
      'MaterialIcons',
      'MaterialCommunityIcons',
      'Octicons',
      'SimpleLineIcons',
      'Zocial',
    ] as const;

    for (const family of families) {
      it(`should render correctly with ${family} family`, () => {
        const { toJSON } = render(
          <Icon name="home" family={family} testID={`icon-${family}-1`} />,
        );
        expect(toJSON()).toMatchSnapshot();
      });
    }
  });

  describe('with different sizes', () => {
    const sizes = ['xs', 'sm', 'md', 'lg', 'xl'] as const;

    for (const size of sizes) {
      it(`should render correctly with ${size} size`, () => {
        const { toJSON } = render(
          <Icon name="home" size={size} testID={`icon-${size}-1`} />,
        );
        expect(toJSON()).toMatchSnapshot();
      });
    }

    it('should render correctly with custom numeric size', () => {
      const { toJSON } = render(
        <Icon name="home" size={32} testID="icon-custom-size-1" />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('with custom styling', () => {
    it('should render correctly with custom color', () => {
      const { toJSON } = render(
        <Icon name="home" color="#ff0000" testID="icon-red-1" />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with custom style', () => {
      const TestComponent: React.FC = () => {
        const styles = useTestStyles();
        return (
          <Icon name="home" style={styles.marginStyle} testID="icon-styled-1" />
        );
      };
      const { toJSON } = render(<TestComponent />);
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('accessibility', () => {
    it('should render correctly with custom accessibility label', () => {
      const { toJSON } = render(
        <Icon
          name="home"
          accessibilityLabel="Custom home icon"
          testID="icon-accessibility-1"
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with accessibility disabled', () => {
      const { toJSON } = render(
        <Icon
          name="home"
          accessible={false}
          testID="icon-no-accessibility-1"
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('interactive functionality', () => {
    it('should render as touchable when onPress is provided', () => {
      const mockOnPress = jest.fn();
      const { toJSON } = render(
        <Icon name="home" onPress={mockOnPress} testID="icon-touchable-1" />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should call onPress when pressed', () => {
      const mockOnPress = jest.fn();
      const { getByTestId } = render(
        <Icon name="home" onPress={mockOnPress} testID="icon-touchable-1" />,
      );

      fireEvent.press(getByTestId('icon-touchable-1-button'));
      expect(mockOnPress).toHaveBeenCalledTimes(1);
    });

    it('should render correctly with children', () => {
      const { toJSON } = render(
        <Icon name="home" testID="icon-with-children-1">
          <Icon
            name="star"
            family="AntDesign"
            size="sm"
            testID="icon-child-1"
          />
        </Icon>,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('edge cases', () => {
    it('should render correctly with unknown family (fallback to MaterialIcons)', () => {
      const { toJSON } = render(
        <Icon
          name="home"
          family={'UnknownFamily' as any}
          testID="icon-unknown-family-1"
        />,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('should render correctly with empty name', () => {
      const { toJSON } = render(<Icon name="" testID="icon-empty-name-1" />);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
