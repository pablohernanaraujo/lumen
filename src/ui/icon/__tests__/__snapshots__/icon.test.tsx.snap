// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon accessibility should render correctly with accessibility disabled 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={false}
  color="#283238"
  name="home"
  size={16}
  testID="icon-no-accessibility-1"
/>
`;

exports[`Icon accessibility should render correctly with custom accessibility label 1`] = `
<MaterialIcons
  accessibilityLabel="Custom home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-accessibility-1"
/>
`;

exports[`Icon edge cases should render correctly with empty name 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons  icon"
  accessible={true}
  color="#283238"
  name=""
  size={16}
  testID="icon-empty-name-1"
/>
`;

exports[`Icon edge cases should render correctly with unknown family (fallback to MaterialIcons) 1`] = `
<MaterialIcons
  accessibilityLabel="UnknownFamily home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-unknown-family-1"
/>
`;

exports[`Icon interactive functionality should render as touchable when onPress is provided 1`] = `
<View
  accessibilityLabel="MaterialIcons home button"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="icon-touchable-1-button"
>
  <MaterialIcons
    accessibilityLabel="MaterialIcons home icon"
    accessible={true}
    color="#283238"
    name="home"
    size={16}
    testID="icon-touchable-1"
  />
</View>
`;

exports[`Icon interactive functionality should render correctly with children 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-with-children-1"
>
  <AntDesign
    accessibilityLabel="AntDesign star icon"
    accessible={true}
    color="#283238"
    name="star"
    size={14}
    testID="icon-child-1"
  />
</MaterialIcons>
`;

exports[`Icon snapshots should render correctly in dark mode 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#FFFFFF"
  name="home"
  size={16}
  testID="icon-1"
/>
`;

exports[`Icon snapshots should render correctly in light mode 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-1"
/>
`;

exports[`Icon with custom styling should render correctly with custom color 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#ff0000"
  name="home"
  size={16}
  testID="icon-red-1"
/>
`;

exports[`Icon with custom styling should render correctly with custom style 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  style={
    {
      "margin": 8,
    }
  }
  testID="icon-styled-1"
/>
`;

exports[`Icon with different families should render correctly with AntDesign family 1`] = `
<AntDesign
  accessibilityLabel="AntDesign home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-AntDesign-1"
/>
`;

exports[`Icon with different families should render correctly with Entypo family 1`] = `
<Entypo
  accessibilityLabel="Entypo home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-Entypo-1"
/>
`;

exports[`Icon with different families should render correctly with EvilIcons family 1`] = `
<EvilIcons
  accessibilityLabel="EvilIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-EvilIcons-1"
/>
`;

exports[`Icon with different families should render correctly with Feather family 1`] = `
<Feather
  accessibilityLabel="Feather home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-Feather-1"
/>
`;

exports[`Icon with different families should render correctly with FontAwesome family 1`] = `
<FontAwesome
  accessibilityLabel="FontAwesome home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-FontAwesome-1"
/>
`;

exports[`Icon with different families should render correctly with FontAwesome5 family 1`] = `
<FontAwesome5
  accessibilityLabel="FontAwesome5 home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-FontAwesome5-1"
/>
`;

exports[`Icon with different families should render correctly with Foundation family 1`] = `
<Foundation
  accessibilityLabel="Foundation home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-Foundation-1"
/>
`;

exports[`Icon with different families should render correctly with Ionicons family 1`] = `
<Ionicons
  accessibilityLabel="Ionicons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-Ionicons-1"
/>
`;

exports[`Icon with different families should render correctly with MaterialCommunityIcons family 1`] = `
<MaterialCommunityIcons
  accessibilityLabel="MaterialCommunityIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-MaterialCommunityIcons-1"
/>
`;

exports[`Icon with different families should render correctly with MaterialIcons family 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-MaterialIcons-1"
/>
`;

exports[`Icon with different families should render correctly with Octicons family 1`] = `
<Octicons
  accessibilityLabel="Octicons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-Octicons-1"
/>
`;

exports[`Icon with different families should render correctly with SimpleLineIcons family 1`] = `
<SimpleLineIcons
  accessibilityLabel="SimpleLineIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-SimpleLineIcons-1"
/>
`;

exports[`Icon with different families should render correctly with Zocial family 1`] = `
<Zocial
  accessibilityLabel="Zocial home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-Zocial-1"
/>
`;

exports[`Icon with different sizes should render correctly with custom numeric size 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={32}
  testID="icon-custom-size-1"
/>
`;

exports[`Icon with different sizes should render correctly with lg size 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={18}
  testID="icon-lg-1"
/>
`;

exports[`Icon with different sizes should render correctly with md size 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={16}
  testID="icon-md-1"
/>
`;

exports[`Icon with different sizes should render correctly with sm size 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={14}
  testID="icon-sm-1"
/>
`;

exports[`Icon with different sizes should render correctly with xl size 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={20}
  testID="icon-xl-1"
/>
`;

exports[`Icon with different sizes should render correctly with xs size 1`] = `
<MaterialIcons
  accessibilityLabel="MaterialIcons home icon"
  accessible={true}
  color="#283238"
  name="home"
  size={12}
  testID="icon-xs-1"
/>
`;
